app/mailers/producer_mailer.rb

```diff
@@ -66,22 +66,4 @@ def total_from_line_items(line_items)
   def tax_total_from_line_items(line_items)
     Spree::Money.new line_items.sum(&:included_tax)
   end
-
-  # This hack makes ActiveRecord skip the default_scope (deleted_at IS NULL)
-  # when eager loading associations. Further details:
-  # https://github.com/rails/rails/issues/11036
-  def with_unscoped_products_and_variants
```

Comment:
@sauloperez I borrowed your hack and extracted it so it could be used elsewhere (I needed it here).

Reply:
I'm fine with the patch now so we deliver this ASAP, but I think we should rather find a long-term solution for the entire app and not only these two associations. Check out https://github.com/openfoodfoundation/openfoodnetwork/commit/04c962432a7664a716907a55a9cd15afd115c6a8 for details on potential more robust solutions. I haven't created an issue yet but this is worth a team discussion.

Personally, I really dislike patching Rails in an app because it gets on the way of future upgrades (it was painful to migrate a Rails 3.2 to 4 with a few custom patches in the past). I would give that gem I mention in the commit a try. The owner is a friend of ours and it seems well maintained.

Reply:
A nicer solution would be very welcome! :smile:

Reply:
Tracked in https://github.com/openfoodfoundation/openfoodnetwork/issues/5287